shim/shim.c

```diff
@@ -15,6 +15,15 @@
  * limitations under the License.
  */
 
+#define _XOPEN_SOURCE 700
+
+/* Unfortunately, OS/X so far only implements issue 6.
+ * O_NOFOLLOW was added by issue 7.
+ */
+#if !defined(O_NOFOLLOW)
+#define _DARWIN_C_SOURCE 1
+#endif
+
 /* Wake vfork exec shim */
 #include <sys/stat.h>
 #include <fcntl.h>
```

Comment:
FWIW, I was able to get it to build by just putting `#include <fcntl.h>` before the `#define _XOPEN_SOURCE 700`

Reply:
Well, that's not an actual fix. By moving the define later, you've essentially removed its affect.